src/test/java/io/confluent/idesidecar/restapi/util/WebClientFactoryTest.java

```diff
@@ -1,20 +1,89 @@
 package io.confluent.idesidecar.restapi.util;
 
 
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static org.junit.Assert.assertSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.matching.UrlPattern;
+import io.confluent.idesidecar.restapi.application.SidecarInfo;
+import io.confluent.idesidecar.restapi.testutil.FakeSideCarProfile;
 import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.vertx.ext.web.client.WebClient;
 import jakarta.inject.Inject;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 @QuarkusTest
+@TestProfile(FakeSideCarProfile.class)
 public class WebClientFactoryTest {
   @Inject WebClientFactory webClientFactory;
+  @Inject SidecarInfo sidecarInfo;
+
+  WireMock wireMock;
+  WireMockServer wireMockServer;
+
+  @BeforeEach
+  public void setup() {
+    wireMockServer = new WireMockServer(options().dynamicPort());
+    wireMockServer.start();
+    WireMock.configureFor("localhost", wireMockServer.port());
+  }
+
+  @Test
+  void getDefaultWebClientOptionsIncludesUserAgent() {
+    var userAgent = webClientFactory.getDefaultWebClientOptions().getUserAgent();
+    assertTrue(userAgent.contains("Confluent-for-VSCode/"));
+  }
+
+
+  @Test
+  public void webClientOptionsShouldContainUserAgentOnCreation() {
+    UrlPattern urlPattern = urlEqualTo("/some-endpoint");
+    // Register the WireMock stub
+    wireMockServer.stubFor(
+        WireMock.get(urlPattern)
+                .willReturn(
+                    aResponse()
+                        .withStatus(200)
+                        .withBody("{\"message\":\"success\"}")
+                )
+    );
+
+    String userAgent = webClientFactory.getDefaultWebClientOptions().getUserAgent();
```

Comment:
Let's add this assertion here as follows (as mentioned by Randall [here](https://github.com/confluentinc/ide-sidecar/pull/202/files#r1870476516))

```suggestion
    String userAgent = webClientFactory.getDefaultWebClientOptions().getUserAgent();

    // Assert that this matches what SidecarInfo tells us the user agent is
    assertEquals(userAgent, sidecarInfo.userAgent());
```